src/controllers/dynakube/activegate/internal/capability/reconciler_test.go

```diff
@@ -1,517 +1,213 @@
-// 				{Name: consts.StatsdContainerName},
-// 			},
-// 			consts.EecContainerName,
-// 			fmt.Sprintf(`Cannot find container "%s" in the provided slice (len 2)`, consts.EecContainerName),
-// 		)
-// 	})
-
-// 	t.Run("happy path", func(t *testing.T) {
-// 		verify(t,
-// 			[]corev1.Container{
-// 				{Name: consts.StatsdContainerName},
-// 			},
-// 			consts.StatsdContainerName,
-// 			"",
-// 		)
-// 	})
-// }
+import (
+	"context"
+	"testing"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/capability"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/internal/authtoken"
+	"github.com/Dynatrace/dynatrace-operator/src/kubesystem"
+	"github.com/Dynatrace/dynatrace-operator/src/scheme"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testToken    = "dt.testtoken.test"
+	testUID      = "test-uid"
+	testDynakube = "test-dynakube"
+)
+
+type MockStatefulSetReconciler struct {
+	mock.Mock
+}
+
+func (m *MockStatefulSetReconciler) Reconcile() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+type MockCustomPropertiesReconciler struct {
+	mock.Mock
+}
+
+func (m *MockCustomPropertiesReconciler) Reconcile() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+func createMultiCapabilityReconciler(t *testing.T) *Reconciler {
+	clt := fake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		WithObjects(&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: kubesystem.Namespace,
+				UID:  testUID,
+			},
+		}).
+		WithObjects(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dynatracev1beta1.AuthTokenSecretSuffix,
+				Namespace: testNamespace,
+			},
+			Data: map[string][]byte{authtoken.ActiveGateAuthTokenName: []byte(testToken)},
+		}).
+		Build()
+	instance := &dynatracev1beta1.DynaKube{
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			ActiveGate: dynatracev1beta1.ActiveGateSpec{
+				Capabilities: []dynatracev1beta1.CapabilityDisplayName{
+					dynatracev1beta1.RoutingCapability.DisplayName,
+					dynatracev1beta1.KubeMonCapability.DisplayName,
+					dynatracev1beta1.MetricsIngestCapability.DisplayName,
+					dynatracev1beta1.DynatraceApiCapability.DisplayName,
+					dynatracev1beta1.SyntheticCapability.DisplayName,
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testDynakube,
+			Namespace: testNamespace,
+		}}
+
+	statefulSetReconcilerMock := MockStatefulSetReconciler{}
+	statefulSetReconcilerMock.On("Reconcile").Return(nil)
+
+	custompropertiesReconcilerMock := MockCustomPropertiesReconciler{}
+	custompropertiesReconcilerMock.On("Reconcile").Return(nil)
```

Comment:
you could use the mocks to check if they were called or not with `.AssertCalled`
(you would need to be able to access these mocks in the actual tests, which in the current form not possible)